### ISSUE TITLE
Errors in Firefox console during import of channel

### ISSUE THREAD

**Issue by cpauya** [learningequality/kolibri#7046]
### Observed behavior
When importing a channel, some errors show in the Firefox's console.  

On the side, I don't know if the above error causes this "131% progress" error coz I cannot reproduce this consistently.  But during import the progress bar  moves back and forth and the values indicate ... 20%, 50%, 25%, 65%,... 90%, 131%, 93%, 100% in order.

I managed to capture this screenshot.
![Screen Shot 2020-06-12 at 8 24 43 PM](https://user-images.githubusercontent.com/175580/84504509-a5148200-acee-11ea-96f6-5a04e8fe729d.png)

**Try 2**
When I tried again, it showed the errors but the progress bar is ok:
![2020-06-12-fri--import channel bug 1](https://user-images.githubusercontent.com/175580/84505526-70092f00-acf0-11ea-8b71-24219543ddde.gif)

**Try 3**
Re-tried again and this time the import did not proceed and showed a "Something went wrong!" page:
![Screen Shot 2020-06-12 at 9 17 20 PM](https://user-images.githubusercontent.com/175580/84506736-5d8ff500-acf2-11ea-958f-9387a67a998a.png)

This is the console error:
> 21:17:31.263 TypeError: An id must be specified api-resource.js:752

Here's the log:
[console-export-2020-6-12_21-21-4.txt](https://github.com/learningequality/kolibri/files/4770996/console-export-2020-6-12_21-21-4.txt)

### Expected behavior
No errors must show during content import.

### User-facing consequences
Import functionality may be broken due to those errors.

### Errors and logs

**Try 1 and 2**
[debug.txt](https://github.com/learningequality/kolibri/files/4770874/debug.txt)
[kolibri.txt](https://github.com/learningequality/kolibri/files/4770875/kolibri.txt)
[console-export-2020-6-12_21-2-48.txt](https://github.com/learningequality/kolibri/files/4770919/console-export-2020-6-12_21-2-48.txt)

### Steps to reproduce

1. Delete the `~/.kolibri/` folder - make a backup if needed.
1. Checkout `release-v0.14.x`.
1. Follow the [Setting up Kolibri for Development](https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#setting-up-kolibri-for-development).
1. Make sure to show the Firefox Console window to notice the errors.
1. Do a Quick Setup of Kolibri, then import **Kolibri QA Channel** from Studio using the token `nakav-mafak`.  You can also do the usual import by selecting resources.
1. Notice the errors shown in the Firefox console.

### Context
* Kolibri version: `release-v0.14.x` branch
* Operating system: macOS v10.14.6 Mojave
* Browser: Firefox v77.0.1
* Python: v3.7.7

**Comment by indirectlylit** [comment]
still present

**Comment by MisRob** [comment]
## 1. "[Vue warn]: Property or method "disableTransition" is not defined on the instance but referenced during render."

This can be fixed by removing [`disableTransition` condition](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/loaders/KCircularLoader.vue#L13) or adding it as a property to _KCircularLoader_.

Edit: I've just noticed that the property is [there already](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/loaders/KCircularLoader.vue#L109) - it seems to be only syntax error[ on the line before](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/loaders/KCircularLoader.vue#L108). I haven't found any instances of _KCircularLoader_ with `disableTransition` set to `true` so I am not sure if we want to use it. If yes, syntax fix should help and if not, we can remove both the condition and the property. Not sure what's better here as I am not familiar with all planned use-cases (@indirectlylit , @nucleogenesis ?)

## 2. [ERROR: kolibri/core/assets/src/api-resource.js] An error occurred Error: Request failed with status code 404

I haven't found out what's the cause of this problem yet though I've found a way to reproduce it. Some observations:

It's not Firefox only issue - reproduced in Chrome as well.

How to reproduce:
1. Remove _~/.kolibri/_ (important, you won't be able to replicate the error if you opened resources selection page for a chosen test channel in the past)
2. Go to [Select resources for import](http://127.0.0.1:8000/en/device/#/content/channels)
3. Click _Select Resources_ for a channel (I am using _Touchable Earth_ in this example)
4. The first call to http://127.0.0.1:8000/device/api/device_channel/000409f81dbe5d1ba67101cb9fed4530/?include_fields=total_resources%2Ctotal_file_size%2Con_device_resources%2Con_device_file_size&1592912211734=1592912211734 triggered by [getChannelWithContentSizes](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/apiChannelMetadata.js#L5) [from here](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/handlers.js#L161) fails on `404` with `[{"id":"NOT_FOUND","metadata":{"view":"Device Channel Metadata Instance"}}]` in response.
5. Shortly after the same call triggered [from here](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/utils.js#L61) is made http://127.0.0.1:8000/device/api/device_channel/000409f81dbe5d1ba67101cb9fed4530/?include_fields=total_resources%2Ctotal_file_size%2Con_device_resources%2Con_device_file_size&1592912214451=1592912214451 but this time it ends with success. This is true for all same subsequent calls (you can try to access the page again). To reproduce `404` again, _~/.kolibri/_ must be removed.

Any ideas about what could be causing this behavior? I checked some backend code around [DeviceChannelMetadataViewSet ](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/api.py#L73) and haven't noticed anything suspicious so far. The fact that only the first call fails and also that there's a need to remove _~/.kolibri/_ to be able to reproduce it makes me wonder if it could be somehow related to server or bundling setup?

## 3. Strange progress bar numbers

I haven't been able to reproduce yet.

**Comment by indirectlylit** [comment]
> we can remove both the condition and the property

yup - and in fact I'd forgotten that I already opened a PR for it here: https://github.com/learningequality/kolibri-design-system/pull/38

**Comment by indirectlylit** [comment]
> Request failed with status code 404

This behavior is reported here: https://github.com/learningequality/kolibri/issues/5343

**Comment by indirectlylit** [comment]
@MisRob would it be possible to add some logging before or after the 404 indicating that this is an expected response and not an error to be concerned about?

**Comment by rtibbles** [comment]
Note the main error here is actual the final one in the second screenshot `TypeError: id must be defined` - here an actual error is occurring as the frontend code is asking the API resource layer to fetch an object with id `undefined`.

**Comment by MisRob** [comment]
@indirectlylit 

**Yes, I think that adding a message [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/handlers.js#L166.) should help. It's my first encounter with tasks and this logic so I am not sure if I can understand the problem completely. I guess this error is not a priority right now as it doesn't break anything for users so just adding a comment below for now and moving on. Do we want to at least add logging as you suggest for 0.14? I can prepare a PR.**

---

~~From #5343 and comments here it seems that `404` is desired and is somehow used for a decision whether channel metadata task should be triggered or not though I can't see this happening anywhere in the code. Wondering if it would be possible to trigger download channel metadata task sooner if needed, e.g. to add a check if we have data available already or not (it seems that we currently use `channelIsOnDevice` getter for that purpose) somewhere before [installedChannelsPromise](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/handlers.js#L161)  and if not, trigger the task instead of handling this later [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/device/assets/src/modules/wizard/handlers.js#L226) and avoid `404` for this scenario?~~

Edit: Oh I see now, it seems that we don't get any information about the availability of the channel in advance from backend, except of getting that 404. Okay, in that case, in makes sense to just add a message.

**Comment by MisRob** [comment]
@rtibbles 

> Note the main error here is actual the final one in the second screenshot TypeError: id must be defined - here an actual error is occurring as the frontend code is asking the API resource layer to fetch an object with id undefined.

Will try to reproduce it again tomorrow together with checking if I can find some hints in a server log. I haven't seen this problem in Firefox yet unfortunately. Loader behavior seemed to be fine too.

**Comment by MisRob** [comment]
Regarding `TypeError: id must be defined` error and the loader behavior, I wasn't able to reproduce any of those after lots of attempts and also haven't noticed anything in logs that would help me with that. There's one problem but doesn't seem to be related taking into account that `TypeError` occurred at 21:17.

```
ERROR 2020-06-12 20:36:03,107 django.request Internal Server Error: /en/user/
Traceback (most recent call last):
  File "/Users/cyril/w/le/kolibri/kolibri/core/device/utils.py", line 26, in get_device_setting
    device_settings = DeviceSettings.objects.get()
  File "/Users/cyril/w/le/kolibri/kolibri/core/device/models.py", line 41, in get
    model = super(DeviceSettingsManager, self).get(**kwargs)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
kolibri.core.device.models.DoesNotExist: DeviceSettings matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/core/handlers/base.py", line 217, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/core/handlers/base.py", line 215, in _get_response
    response = response.render()
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/response.py", line 107, in render
    self.content = self.rendered_content
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/response.py", line 84, in rendered_content
    content = template.render(context, self._request)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/backends/django.py", line 66, in render
    return self.template.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 207, in render
    return self._render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/loader_tags.py", line 72, in render
    result = block.nodelist.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/Users/cyril/.local/share/virtualenvs/kolibri-cG-2Tn9P/lib/python3.7/site-packages/django/template/library.py", line 203, in render
    output = self.func(*resolved_args, **resolved_kwargs)
  File "/Users/cyril/w/le/kolibri/kolibri/core/webpack/templatetags/webpack_tags.py", line 42, in webpack_asset
    return hook.render_to_page_load_sync_html()
  File "/Users/cyril/w/le/kolibri/kolibri/core/webpack/hooks.py", line 384, in render_to_page_load_sync_html
    + list(self.js_and_css_tags())
  File "/Users/cyril/w/le/kolibri/kolibri/core/webpack/hooks.py", line 289, in plugin_data_tag
    if self.plugin_data:
  File "/Users/cyril/w/le/kolibri/kolibri/plugins/user/kolibri_plugin.py", line 42, in plugin_data
    "allowGuestAccess": get_device_setting("allow_guest_access"),
  File "/Users/cyril/w/le/kolibri/kolibri/core/device/utils.py", line 33, in get_device_setting
    raise DeviceNotProvisioned
kolibri.core.device.utils.DeviceNotProvisioned
```

**Comment by indirectlylit** [comment]
the `disableTransition` error should finally be resolved by #7188

@MisRob would you mind filing new issues for the remaining relevant parts and closing this issue?

**Comment by MisRob** [comment]
@indirectlylit Sure. Here are the issues: #7197, #7196 (Could you please add labels? I am not sure what priority are they) and a [PR containing 404 logging](https://github.com/learningequality/kolibri/pull/7195) as it was faster than opening a new issue :)
